premailer/tests/test_premailer.py

```diff
@@ -2302,3 +2302,28 @@ def test_turnoff_cache_works_as_expected(self):
         result_html = p.transform()
 
         compare_html(expect_html, result_html)
+
+    def test_links_without_protocol(self):
+        """If you the base URL is set to https://example.com and your html
+        contains <a href="//otherdomain.com/">...</a> then the URL to point to
+        is "https://otherdomain.com/" not "https://example.com/file.css"
+        """
+        html = """<html>
+        <head>
+        </head>
+        <body>
+        <a href="//example.com">Link</a>
```

Comment:
Hmm... It makes me wonder. Why ever allow or bother with links like this?
The `(href|src)="//...` trick is "reserved" for inclusion of static assets (`script`, `link`, `img`, etc). That's because you'll want to load multiple requests under one page view and when you do that the browser dictates that you'll want them also loaded on HTTPS if you're currently on and HTTPS page. 

However, if you have a link that's not going to be part of the page view. It's for something entirely different later. 

If you want to make a link I think the right thing to do is something like `<a href="otherpage.html">` or `<a href="/relative/abs.html">` (or `<a href="http://specificdomain.com/index.html">` of course). 

I'm sorry if I'm making your life hard here :) 
But I'm not convinced premailer should get itself involved in (bad) HTML choices. Perhaps I'm wrong about using `//` in hyperlinks and I'm more than happy to be educated and convinced.